app/utils.py

```diff
@@ -171,19 +180,283 @@ def lazy_format(s, f):
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+
+class GenericPass:
+    """Class for creating and managing Generic passes in Google Wallet.
+
+    Attributes:
+        key_file_path: Path to service account key file from Google Cloud
+            Console. Environment variable: GOOGLE_APPLICATION_CREDENTIALS.
+        base_url: Base URL for Google Wallet API requests.
+    """
+
+    def __init__(self):
+        self.key_file_path = os.environ.get(
+            "GOOGLE_WALLET_APPLICATION_CREDENTIALS", "/path/to/key.json"
+        )
+        self.base_url = "https://walletobjects.googleapis.com/walletobjects/v1"
+        self.batch_url = "https://walletobjects.googleapis.com/batch"
+        self.class_url = f"{self.base_url}/genericClass"
+        self.object_url = f"{self.base_url}/genericObject"
+
+        # Set up authenticated client
+        self.auth()
+
+    # [END setup]
+
+    # [START auth]
+    def auth(self):
+        """Create authenticated HTTP client using a service account file."""
+        self.credentials = Credentials.from_service_account_file(
+            self.key_file_path,
+            scopes=["https://www.googleapis.com/auth/wallet_object.issuer"],
+        )
+
+        self.http_client = AuthorizedSession(self.credentials)
+
+    # [END auth]
+
+    # [START createObject]
+    def create_object(
+        self,
+        issuer_id: str,
+        class_suffix: str,
+        object_suffix: str,
+        cardObject: dict,
+        qrValue: str,
+    ) -> str:
+        """Create an object.
+
+        Args:
+            issuer_id (str): The issuer ID being used for this request.
+            class_suffix (str): Developer-defined unique ID for the pass class.
+            object_suffix (str): Developer-defined unique ID for the pass object.
+
+        Returns:
+            The pass object ID: f"{issuer_id}.{object_suffix}"
+        """
+
+        # Check if the object exists
+        response = self.http_client.get(
+            url=f"{self.object_url}/{issuer_id}.{object_suffix}"
+        )
+
+        if response.status_code == 200:
+            print(
+                f"[GOOGLE_WALLET]: Object {issuer_id}.{object_suffix} already exists!"
+            )
+            # print(response.text)
+            return f"{issuer_id}.{object_suffix}"
+        elif response.status_code != 404:
+            # Something else went wrong...
+            print("[GOOGLE_WALLET]:", response.text)
+            return f"{issuer_id}.{object_suffix}"
+
```

Comment:
El 404 es gestiona solito? O faltaria un else?

Reply:
In reality, it's how we determine if the object does exist or not, if we receive a 404 we then create it by dong a POST request instead of a GET one. But thanks for the question, it made me reorganize the code to make it easier to understand.